### PR TITLE
#2 Add saving goal definition to data model

### PR DESCRIPTION
**Pull request by AplusKminus**

### CODE REVIEW

File: app/src/main/kotlin/app/pmsoft/ispork/data/SavingGoalDao.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package app.pmsoft.ispork.data
+
+import androidx.room.*
+
+@Dao
+interface SavingGoalDao {
+
+  @Query("SELECT * FROM savingGoal ORDER BY name")
```
Comment (0plusKminus) [inline app/src/main/kotlin/app/pmsoft/ispork/data/SavingGoalDao.kt]: Is ordering by name sensible?
I believe ordering by category or by target date might be more sensible.

File: app/src/main/kotlin/app/pmsoft/ispork/data/SavingGoalDao.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package app.pmsoft.ispork.data
```
Comment (0plusKminus) [inline app/src/main/kotlin/app/pmsoft/ispork/data/SavingGoalDao.kt]: A get by category method would be useful.

File: app/src/main/kotlin/app/pmsoft/ispork/data/SavingGoalDao.kt (hunk 1 above)
Comment (AplusKminus) [reply app/src/main/kotlin/app/pmsoft/ispork/data/SavingGoalDao.kt]: I agree, see the updated commit.

File: app/src/main/kotlin/app/pmsoft/ispork/data/SavingGoalDao.kt (hunk 2 above)
Comment (AplusKminus) [reply app/src/main/kotlin/app/pmsoft/ispork/data/SavingGoalDao.kt]: True. I added the method. Good job spotting that!

**Pull request merged**